app/services/wish_service.rb

```diff
@@ -71,6 +75,25 @@ def num_wishlist_deal(deal_id)
       num
     end
 
+    # Get array of user id who wishlist the venues associated with a deal
+    def get_user_id(deal_id, activate_date = nil)
+      deal_venue = DealVenue.where(:deal_id => deal_id)
+      user_id = []
+      deal_venue.each do |dv|
+        if activate_date == nil
+          wishes = Wish.where(:venue_id => dv.venue_id)
+        else
+          wishes = Wish.where(:venue_id => dv.venue_id).where('created_at <= ?',activate_date)
+        end
+        user_id.each do |u|
+          wishes = wishes.reject{|r| r.user_id == u}
```

Comment:
Isn't it better to add a distinct query in line 84 and 86 to find unique user_id then to use this reject method?